sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -77,34 +78,20 @@ public boolean hasStarted() {
     public void startPushNotifications(String deviceRegistrationToken, Consumer<Throwable> errorHandler) {
         this.deviceRegistrationToken = deviceRegistrationToken;
         this.registrationErrorHandler = errorHandler;
-
+        Log.v("device_id: ", deviceRegistrationToken);
         if (this.isPushNotificationsStarted) {
             return;
         }
-
-        String skypeUserToken;
-        try {
-            skypeUserToken = communicationTokenCredential.getToken().get().getToken();
-        } catch (ExecutionException | InterruptedException e) {
-            throw logger.logExceptionAsError(
-                new RuntimeException("Get skype user token failed for push notification: " + e.getMessage()));
-        }
-
-        try {
-            this.refreshEncryptionKeys();
-            this.registrarClient.register(skypeUserToken, deviceRegistrationToken, this.cryptoKey, this.authKey);
-        } catch (Throwable throwable) {
-            this.logger.error("Failed to start push notifications!");
-            errorHandler.accept(throwable);
-            return;
-        }
+        this.workManager = WorkManager.getInstance();
+        this.registrationDataContainer = RegistrationDataContainer.instance();
+        stopPushNotifications();
 
         this.isPushNotificationsStarted = true;
         this.pushNotificationListeners.clear();
         this.logger.info("Successfully started push notifications!");
-
-        long delayInMS = 1000L * (long) (Integer.parseInt(PUSHNOTIFICATION_REGISTRAR_SERVICE_TTL) - 30);
-        this.scheduleRegistrationRenew(delayInMS, 0);
+        // 15 minutes is the minimum interval worker manager allows
+        long delay = 15;
+        this.startRegistrationRenewalWorker(delay);
```

Comment:
I assume that a registration request will be sent immediately and then it will wait for 15mins before the next renewal, right? in this case calling it "delay" could be confusing. "interval" might be a better name.

Reply:
Yes. It is interval. The first execution starts immediately.